src/libraries/Silk.Net.Windowing/Abstract/IDesktopSurface.cs

```diff
@@ -0,0 +1,150 @@
+﻿using Silk.NET.Core;
+using Silk.NET.Maths;
+
+namespace Silk.NET.Windowing
+{
+    /// <summary>
+    /// A surface which wraps a Desktop Window.
+    /// </summary>
+    public interface IDesktopSurface : ISurface
+    {
+        /// <summary>
+        /// Whether or not the window is visible.
+        /// </summary>
+        bool IsVisible { get; set; }
+
+        /// <summary>
+        /// The position of the window. If set to -1, use the backend default.
+        /// </summary>
+        Vector2D<int> Position { get; set; }
+
+        /// <summary>
+        /// The size of the window in pixels.
+        /// </summary>
+        new Vector2D<int> Size { get; set; }
+
+        /// <summary>
+        /// The window title.
+        /// </summary>
+        string Title { get; set; }
+
+        /// <summary>
+        /// The window state.
+        /// </summary>
+        WindowState WindowState { get; set; }
+
+        /// <summary>
+        /// The window border.
+        /// </summary>
+        WindowBorder WindowBorder { get; set; }
+
+        /// <summary>
+        /// The video mode.
+        /// </summary>
+        VideoMode VideoMode { get; set; }
+    
+        /// <summary>
+        /// Gets the screen on which this window is active.
+        /// </summary>
+        IScreen? CurrentScreen { get; set; }
+
+        /// <summary>
+        /// Gets the available screens for this surface.
+        /// </summary>
+        IEnumerable<IScreen>? AvailableScreens { get; }
+
+        /// <summary>
+        /// Gets or sets whether the window waits for an event to be posted before existing <see cref="DoEvents" />.
+        /// </summary>
+        bool IsEventDriven { get; set; }
+
+        /// <summary>
+        /// Gets or sets whether the window has been requested to close.
+        /// </summary>
+        bool IsCloseRequested { get; set; }
+
+        /// <summary>
+        /// Gets whether the window is focused or not.
+        /// </summary>
+        bool IsFocused { get; }
+
+        /// <summary>
+        /// Gets the distances in screen coordinates from the edges of the content area to the corresponding edges of
+        /// the full window.
+        /// </summary>
+        /// <remarks>
+        /// Because these are distances and not coordinates, they are always zero or positive.
+        /// </remarks>
+        /// <seealso cref="WindowExtensions.GetFullSize"/>
+        Rectangle<int> BorderSize { get; }
```

Comment:
I have no idea what it is.

Reply:
The content area describes the space a window can actually take up. So on a 2160p monitor, it will be `3840 x (2160 - <size of taskbar>)` if that makes any sense.